homeassistant/components/tellduslive.py

```diff
@@ -46,57 +52,161 @@
 ATTR_LAST_UPDATED = 'time_last_updated'
 
 
-def setup(hass, config):
+def setup(hass, config, session=None):
     """Set up the Telldus Live component."""
-    client = TelldusLiveClient(hass, config)
+    from tellduslive import Session, supports_local_api
+    config_filename = hass.config.path(TELLLDUS_CONFIG_FILE)
+    conf = load_json(config_filename)
+
+    def request_configuration(host=None):
+        """Request TelldusLive authorization."""
+        configurator = hass.components.configurator
+        hass.data.setdefault(KEY_CONFIG, {})
+        data_key = host or DOMAIN
+
+        # Configuration already in progress
+        if hass.data[KEY_CONFIG].get(data_key):
+            return
+
+        _LOGGER.info('Configuring TelldusLive %s',
+                     'local client: {}'.format(host) if host else
+                     'cloud service')
+
+        session = Session(public_key=PUBLIC_KEY,
+                          private_key=NOT_SO_PRIVATE_KEY,
+                          host=host,
+                          application=APPLICATION_NAME)
+
+        auth_url = session.authorize_url
+        if not auth_url:
+            _LOGGER.warning('Failed to retrieve authorization URL')
+            return
+
+        _LOGGER.debug('Got authorization URL %s', auth_url)
+
+        def configuration_callback(callback_data):
+            """Handle the submitted configuration."""
+            session.authorize()
+            res = setup(hass, config, session)
+            if not res:
+                hass.async_add_job(configurator.notify_errors,
+                                   hass.data[KEY_CONFIG].get(data_key),
+                                   'Unable to connect.')
+                return
+
+            @asyncio.coroutine
+            def success():
+                """Set up was successful."""
+                conf.update(
+                    {host: {CONF_HOST: host,
+                            CONF_TOKEN: session.access_token}} if host else
+                    {DOMAIN: {CONF_TOKEN: session.access_token,
+                              CONF_TOKEN_SECRET: session.access_token_secret}})
+                save_json(config_filename, conf)
+                hass.async_add_job(configurator.request_done,
```

Comment:
For a device that supports the local API, on https://github.com/home-assistant/home-assistant/pull/10435/files#diff-dd01cf62b52b2972c7c453cffe5e35e6R150 we offer both configuration of the local API as well as the cloud service. In case the user chooses to configure the local API, we hav to close _both_ configurators here, right?